File: source/clibCore/Models/Entity/TemplateDomainRuleModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses
Imports Contensive.Core.Controllers
Imports Newtonsoft.Json

Namespace Contensive.Core.Models.Entity
    Public Class TemplateDomainRuleModel
        Inherits baseModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Template Domain Rules"
        Public Const contentTableName As String = "ccTemplateDomainRules"
        Private Shadows Const contentDataSource As String = "default"
        '
        '====================================================================================================
        ' -- instance properties
        Public Property domainId As Integer
        Public Property templateId As Integer
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass) As TemplateDomainRuleModel
            Return add(Of TemplateDomainRuleModel)(cpCore)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass, ByRef callersCacheNameList As List(Of String)) As TemplateDomainRuleModel
            Return add(Of TemplateDomainRuleModel)(cpCore, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer) As TemplateDomainRuleModel
            Return create(Of TemplateDomainRuleModel)(cpCore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer, ByRef callersCacheNameList As List(Of String)) As TemplateDomainRuleModel
            Return create(Of TemplateDomainRuleModel)(cpCore, recordId, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String) As TemplateDomainRuleModel
            Return create(Of TemplateDomainRuleModel)(cpCore, recordGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String, ByRef callersCacheNameList As List(Of String)) As TemplateDomainRuleModel
            Return create(Of TemplateDomainRuleModel)(cpCore, recordGuid, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String) As TemplateDomainRuleModel
            Return createByName(Of TemplateDomainRuleModel)(cpCore, recordName)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String, ByRef callersCacheNameList As List(Of String)) As TemplateDomainRuleModel
            Return createByName(Of TemplateDomainRuleModel)(cpCore, recordName, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub save(cpCore As coreClass)
            MyBase.save(cpCore)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, recordId As Integer)
            delete(Of TemplateDomainRuleModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, ccGuid As String)
            delete(Of TemplateDomainRuleModel)(cpCore, ccGuid)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String, callersCacheNameList As List(Of String)) As List(Of TemplateDomainRuleModel)
            Return createList(Of TemplateDomainRuleModel)(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String) As List(Of TemplateDomainRuleModel)
            Return createList(Of TemplateDomainRuleModel)(cpCore, sqlCriteria, sqlOrderBy)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String) As List(Of TemplateDomainRuleModel)
            Return createList(Of TemplateDomainRuleModel)(cpCore, sqlCriteria)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub invalidatePrimaryCache(cpCore As coreClass, recordId As Integer)
            invalidateCacheSingleRecord(Of TemplateDomainRuleModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, recordId As Integer) As String
            Return baseModel.getRecordName(Of TemplateDomainRuleModel)(cpcore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, ccGuid As String) As String
            Return baseModel.getRecordName(Of TemplateDomainRuleModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordId(cpcore As coreClass, ccGuid As String) As Integer
            Return baseModel.getRecordId(Of TemplateDomainRuleModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createDefault(cpcore As coreClass) As TemplateDomainRuleModel
            Return createDefault(Of TemplateDomainRuleModel)(cpcore)
        End Function
    End Class
End Namespace

File: source/clibCoreCs/Models/Entity/TemplateDomainRuleModel.cs
﻿//========================================================================
// This conversion was produced by the Free Edition of
// Instant C# courtesy of Tangible Software Solutions.
// Order the Premium Edition at https://www.tangiblesoftwaresolutions.com
//========================================================================

using System;
using System.Collections.Generic;
using System.Text;
using Contensive.BaseClasses;
using Contensive.Core.Controllers;
using Newtonsoft.Json;

namespace Contensive.Core.Models.Entity
{
	public class TemplateDomainRuleModel : baseModel
	{
		//
		//====================================================================================================
		//-- const
		public const string contentName = "Template Domain Rules";
		public const string contentTableName = "ccDomainTemplateRules";
		private const string contentDataSource = "default";
		//
		//====================================================================================================
		// -- instance properties
		public int domainId {get; set;}
		public int templateId {get; set;}
		//
		//====================================================================================================
		public static TemplateDomainRuleModel add(coreClass cpCore)
		{
			return add<TemplateDomainRuleModel>(cpCore);
		}
		//
		//====================================================================================================
		public static TemplateDomainRuleModel add(coreClass cpCore, ref List<string> callersCacheNameList)
		{
			return add<TemplateDomainRuleModel>(cpCore, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public static TemplateDomainRuleModel create(coreClass cpCore, int recordId)
		{
			return create<TemplateDomainRuleModel>(cpCore, recordId);
		}
		//
		//====================================================================================================
		public static TemplateDomainRuleModel create(coreClass cpCore, int recordId, ref List<string> callersCacheNameList)
		{
			return create<TemplateDomainRuleModel>(cpCore, recordId, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public static TemplateDomainRuleModel create(coreClass cpCore, string recordGuid)
		{
			return create<TemplateDomainRuleModel>(cpCore, recordGuid);
		}
		//
		//====================================================================================================
		public static TemplateDomainRuleModel create(coreClass cpCore, string recordGuid, ref List<string> callersCacheNameList)
		{
			return create<TemplateDomainRuleModel>(cpCore, recordGuid, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public static TemplateDomainRuleModel createByName(coreClass cpCore, string recordName)
		{
			return createByName<TemplateDomainRuleModel>(cpCore, recordName);
		}
		//
		//====================================================================================================
		public static TemplateDomainRuleModel createByName(coreClass cpCore, string recordName, ref List<string> callersCacheNameList)
		{
			return createByName<TemplateDomainRuleModel>(cpCore, recordName, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public void save(coreClass cpCore)
		{
			base.save(cpCore);
		}
		//
		//====================================================================================================
		public static void delete(coreClass cpCore, int recordId)
		{
			delete<TemplateDomainRuleModel>(cpCore, recordId);
		}
		//
		//====================================================================================================
		public static void delete(coreClass cpCore, string ccGuid)
		{
			delete<TemplateDomainRuleModel>(cpCore, ccGuid);
		}
		//
		//====================================================================================================
		public static List<TemplateDomainRuleModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy, List<string> callersCacheNameList)
		{
			return createList<TemplateDomainRuleModel>(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList);
		}
		//
		//====================================================================================================
		public static List<TemplateDomainRuleModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy)
		{
			return createList<TemplateDomainRuleModel>(cpCore, sqlCriteria, sqlOrderBy);
		}
		//
		//====================================================================================================
		public static List<TemplateDomainRuleModel> createList(coreClass cpCore, string sqlCriteria)
		{
			return createList<TemplateDomainRuleModel>(cpCore, sqlCriteria);
		}
		//
		//====================================================================================================
		public void invalidatePrimaryCache(coreClass cpCore, int recordId)
		{
			invalidateCacheSingleRecord<TemplateDomainRuleModel>(cpCore, recordId);
		}
		//
		//====================================================================================================
		public static string getRecordName(coreClass cpcore, int recordId)
		{
			return baseModel.getRecordName<TemplateDomainRuleModel>(cpcore, recordId);
		}
		//
		//====================================================================================================
		public static string getRecordName(coreClass cpcore, string ccGuid)
		{
			return baseModel.getRecordName<TemplateDomainRuleModel>(cpcore, ccGuid);
		}
		//
		//====================================================================================================
		public static int getRecordId(coreClass cpcore, string ccGuid)
		{
			return baseModel.getRecordId<TemplateDomainRuleModel>(cpcore, ccGuid);
		}
		//
		//====================================================================================================
		public static TemplateDomainRuleModel createDefault(coreClass cpcore)
		{
			return createDefault<TemplateDomainRuleModel>(cpcore);
		}
	}
}
